operator/operator.go

```diff
@@ -20,37 +20,43 @@ import (
 //         AddFunc:    addFunc,
 //     }
 //
-func ProcessCreate(obj interface{}, operator Operator) error {
-	currentState, err := operator.GetCurrentState(obj)
-	if err != nil {
-		return microerror.MaskAny(err)
+func ProcessCreate(obj interface{}, resources ...Resource) error {
```

Comment:
We recently introduced the framework of how operators should reconcile current state to desired state. Since this was a good first step, we realized reconciliation should be done inside some specific scope according to the resource being reconciled. This PR makes the framework accept multiple resources which improves how reconcilable resources can be implemented.